plugins/htpasswd/src/htpasswd.ts

```diff
@@ -13,48 +15,68 @@ import {
   sanityCheck,
 } from './utils';
 
-export interface VerdaccioConfigApp extends Config {
+export type HTPasswdConfig = {
   file: string;
+  algorithm?: HtpasswdHashAlgorithm;
+  rounds?: number;
+  max_users?: number;
   slow_verify_ms?: number;
-}
+} & Config;
 
+export const DEFAULT_BCRYPT_ROUNDS = 10;
 export const DEFAULT_SLOW_VERIFY_MS = 200;
 
 /**
  * HTPasswd - Verdaccio auth class
  */
-export default class HTPasswd implements IPluginAuth<VerdaccioConfigApp> {
+export default class HTPasswd implements IPluginAuth<HTPasswdConfig> {
   /**
    *
    * @param {*} config htpasswd file
    * @param {object} stuff config.yaml in object from
    */
   private users: {};
-  private stuff: {};
-  private config: {};
-  private verdaccioConfig: Config;
   private maxUsers: number;
   private path: string;
+  private hashConfig: HtpasswdHashConfig;
   private slowVerifyMs: number;
   private logger: Logger;
   private lastTime: any;
   // constructor
-  public constructor(config: AuthConf, stuff: VerdaccioConfigApp) {
+  public constructor(config: AuthConf, options: PluginOptions<HTPasswdConfig>) {
     this.users = {};
 
-    // config for this module
-    this.config = config;
-    this.stuff = stuff;
-
     // verdaccio logger
-    this.logger = stuff.logger;
-
-    // verdaccio main config object
-    this.verdaccioConfig = stuff.config;
+    this.logger = options.logger;
 
     // all this "verdaccio_config" stuff is for b/w compatibility only
     this.maxUsers = config.max_users ? config.max_users : Infinity;
 
+    let algorithm: HtpasswdHashAlgorithm;
+    let rounds: number | undefined;
+
+    if (config.algorithm === undefined) {
+      // to avoid breaking changes we uses crypt, future version
+      // of this plugin uses bcrypt by default
+      // https://github.com/verdaccio/verdaccio/pull/2072#issuecomment-770235502
+      algorithm = HtpasswdHashAlgorithm.crypt;
```

Comment:
I suggest adding warning about this upcoming breaking change.

For example:
```
this.logger.warn('Using deprecated "crypt" algorithm that will be replaced by "bcrypt" in the next major version of the plugin. Read more: https://github.com/verdaccio/monorepo/pull/580');
```

Reply:
Good call.